internal/monitors/elasticsearch/elasticsearch.go

```diff
@@ -319,6 +319,139 @@ func (m *Monitor) sendDatapoints(dps []*datapoint.Datapoint) {
 	}
 }
 
+// GetExtraMetrics returns additional metrics to allow through.
+func (c *Config) GetExtraMetrics() []string {
+	var extraMetrics []string
+	if c.EnableEnhancedClusterHealthStats {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupCluster]...)
+	}
+	if c.EnableEnhancedHTTPStats {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupNodeHTTP]...)
+	}
+	if c.EnableEnhancedJVMStats {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupNodeJvm]...)
+	}
+	if c.EnableEnhancedProcessStats {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupNodeProcess]...)
+	}
+	if c.EnableEnhancedThreadPoolStats {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupNodeThreadPool]...)
+	}
+	if c.EnableEnhancedTransportStats {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupNodeTransport]...)
+	}
+	enhancedStatsForIndexGroups := utils.StringSliceToMap(append(c.EnableEnhancedNodeStatsForIndexGroups, c.EnableEnhancedIndexStatsForIndexGroups...))
+	if enhancedStatsForIndexGroups[client.StoreStatsGroup] {
+		extraMetrics = append(extraMetrics, elasticsearchIndicesStoreThrottleTime)
+	}
+	if enhancedStatsForIndexGroups[client.IndexingStatsGroup] {
+		extraMetrics = append(extraMetrics, elasticsearchIndicesIndexingIndexCurrent)
```

Comment:
Do these not relate to the groups somehow? We ideally wouldn't have to hardcode so many metrics here.